supabase/functions/updates/index.ts

```diff
@@ -3,20 +3,34 @@ import { serve } from 'https://deno.land/std@0.200.0/http/server.ts'
 
 import { getRedis } from '../_utils/redis.ts'
 import { update } from '../_utils/update.ts'
-import { methodJson, sendRes, sendResText } from '../_utils/utils.ts'
+import { methodJson, sendRes, sendResText, reverseDomainRegex, deviceIdRegex } from '../_utils/utils.ts'
 import type { AppInfos, BaseHeaders } from '../_utils/types.ts'
 
 const APP_DOES_NOT_EXIST = { message: 'App not found', error: 'app_not_found' }
 const APP_VERSION_NO_NEW = { message: 'No new version available' }
 const CACHE_NO_NEW_VAL = 'NO_NEW'
 
 const jsonRequestSchema = z.object({
-  device_id: z.string(),
-  version_name: z.string(),
-  app_id: z.string(),
+  app_id: z.string({
+    required_error: "App ID is required",
+    invalid_type_error: "App ID name must be a string",
+  }),
+  device_id: z.string({
+    required_error: "Device ID is required",
+    invalid_type_error: "Device ID must be a string",
+  }).max(36),
+  version_name: z.string({
+    required_error: "Version name is required",
+    invalid_type_error: "Version name must be a string",
+  }),
   is_emulator: z.optional(z.boolean()),
-  is_prod: z.optional(z.boolean()),
+  is_prod: z.optional(z.boolean())
```

Comment:
I made a mistake while working on this zod schema, if you want you can port the fixed version from [here](https://github.com/Cap-go/capgo/pull/387) or wait until that PR is merged and resolve merge conflicts